src/FakeItEasy/Creation/DelegateProxies/CastleInvocationCallAdapter.cs

```diff
@@ -0,0 +1,98 @@
+namespace FakeItEasy.Creation.DelegateProxies
+{
+    using System;
+    using System.Diagnostics;
+    using System.Reflection;
+    using Castle.DynamicProxy;
+    using FakeItEasy.Configuration;
+    using FakeItEasy.Core;
+
+    /// <summary>
+    /// An adapter that adapts an <see cref="IInvocation" /> to a <see cref="IFakeObjectCall" />.
+    /// </summary>
+    internal class CastleInvocationCallAdapter

```

Comment:
Maybe `CastleInvocationDelegateCallAdapter`, since it seems to be specific to delegates?